splice_machine/src/main/java/com/splicemachine/derby/impl/sql/catalog/SpliceDataDictionary.java

```diff
@@ -1692,4 +1693,50 @@ public void addMinRetentionPeriodColumn(TransactionController tc) throws Standar
                     SYSTABLESRowFactory.MIN_RETENTION_PERIOD));
         }
     }
+
+    public void setJavaClassNameColumnInSysAliases(TransactionController tc) throws StandardException {
+        SchemaDescriptor sd = getSystemSchemaDescriptor();
+        TabInfoImpl ti=getNonCoreTI(SYSALIASES_CATALOG_NUM);
+        faultInTabInfo(ti);
+
+        FormatableBitSet columnToReadSet=new FormatableBitSet(SYSALIASESRowFactory.SYSALIASES_COLUMN_COUNT);
+        FormatableBitSet columnToUpdateSet=new FormatableBitSet(SYSALIASESRowFactory.SYSALIASES_COLUMN_COUNT);
+        for(int i=0;i<SYSALIASESRowFactory.SYSALIASES_COLUMN_COUNT;i++){
+            // we do not want to read the saved serialized plan
+            if (i+1 == SYSALIASESRowFactory.SYSALIASES_JAVACLASSNAME) {
+                columnToReadSet.set(i);
+            }
+            columnToUpdateSet.set(i);
+        }
+        /* Set up a couple of row templates for fetching CHARS */
+        DataValueDescriptor[] rowTemplate = new DataValueDescriptor[SYSALIASESRowFactory.SYSALIASES_COLUMN_COUNT];
+        DataValueDescriptor[] replaceRow= new DataValueDescriptor[SYSALIASESRowFactory.SYSALIASES_COLUMN_COUNT];
+
+        /* Scan the entire heap */
+        ScanController sc=
+                tc.openScan(
+                        ti.getHeapConglomerate(),
+                        false,
+                        TransactionController.OPENMODE_FORUPDATE,
+                        TransactionController.MODE_TABLE,
+                        TransactionController.ISOLATION_REPEATABLE_READ,
+                        columnToReadSet,
+                        null,
+                        ScanController.NA,
+                        null,
+                        null,
+                        ScanController.NA);
+
+        while(sc.fetchNext(rowTemplate)){
+            /* If JAVACLASSNAME was set to null, rewrite it to "NULL" string literal instead. */
+            for (int i=0; i<rowTemplate.length; i++) {
+                if (i+1 == SYSALIASESRowFactory.SYSALIASES_JAVACLASSNAME)
+                    if(replaceRow[i] == null) {
```

Comment:
We need to check `rowTemplate[I] == null` here which has the fetched row content.

Reply:
Thanks!